tests/apps/pubsub-subscriber_grpc/app.go

```diff
@@ -243,29 +351,55 @@ func (s *server) OnTopicEvent(ctx context.Context, in *runtimev1pb.TopicEventReq
 			Status: runtimev1pb.TopicEventResponse_DROP, //nolint:nosnakecase
 		}, nil
 	}
-
+	log.Printf("(%s) data %s and the content type (%s)", reqID, in.Data, in.DataContentType)
 	var msg string
-	err := json.Unmarshal(in.Data, &msg)
-	if err != nil {
-		log.Printf("(%s) Responding with DROP. Error while unmarshaling JSON data: %v", reqID, err)
-		// Return success with DROP status to drop message
-		return &runtimev1pb.TopicEventResponse{
-			Status: runtimev1pb.TopicEventResponse_DROP, //nolint:nosnakecase
-		}, err
-	}
-
-	// Raw data does not have content-type, so it is handled as-is.
-	// Because the publisher encodes to JSON before publishing, we need to decode here.
-	if strings.HasPrefix(in.Topic, pubsubRaw) {
-		var actualMsg string
-		err = json.Unmarshal([]byte(msg), &actualMsg)
+	var err error
+	if !strings.Contains(in.Topic, "bulk") {
+		// This is the old flow where always the content type is application/json
+		// and data is always json serialized
+		err = json.Unmarshal(in.Data, &msg)
 		if err != nil {
-			log.Printf("(%s) Error extracing JSON from raw event: %v", reqID, err)
-		} else {
-			msg = actualMsg
+			log.Printf("(%s) Responding with DROP. Error while unmarshaling JSON data: %v", reqID, err)
+			// Return success with DROP status to drop message
+			return &runtimev1pb.TopicEventResponse{
+				Status: runtimev1pb.TopicEventResponse_DROP, //nolint:nosnakecase
+			}, err
+		}
+		if strings.HasPrefix(in.Topic, pubsubRaw) {
+			var actualMsg string
+			err = json.Unmarshal([]byte(msg), &actualMsg)
+			if err != nil {
+				log.Printf("(%s) Error extracing JSON from raw event: %v", reqID, err)
+			} else {
+				msg = actualMsg
+			}
+		}
```

Comment:
Reviewers : 
Previous comment ... 
```// Raw data does not have content-type, so it is handled as-is.
	// Because the publisher encodes to JSON before publishing, we need to decode here.```